pkg/asset/manifests/aws/cluster.go

```diff
@@ -151,19 +152,62 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 				PresignedURLDuration: &metav1.Duration{Duration: 1 * time.Hour},
 			},
 			ControlPlaneLoadBalancer: &capa.AWSLoadBalancerSpec{
-				Name:             ptr.To(clusterID.InfraID + "-ext"),
+				Name:             ptr.To(clusterID.InfraID + "-int"),
 				LoadBalancerType: capa.LoadBalancerTypeNLB,
-				Scheme:           &capa.ELBSchemeInternetFacing,
+				Scheme:           &capa.ELBSchemeInternal,
 				AdditionalListeners: []capa.AdditionalListenerSpec{
 					{
 						Port:     22623,
 						Protocol: capa.ELBProtocolTCP,
 					},
 				},
+				IngressRules: []capa.IngressRule{
+					{
+						Description: "Machine Config Server internal traffic from cluster",
+						Protocol:    capa.SecurityGroupProtocolTCP,
+						FromPort:    22623,
+						ToPort:      22623,
+						CidrBlocks:  []string{mainCIDR.String()},
+					},
+				},
 			},
 		},
 	}
 
+	if installConfig.Config.Publish == types.ExternalPublishingStrategy {
+		// FIXME: CAPA bug. Remove when fixed upstream
+		// The primary and secondary load balancers in CAPA share the same
+		// security group. However, specifying an ingress rule only in the
+		// second LB does not seem to take effect, forcing us to add it to the
+		// primary LB instead.
```

Comment:
Is there a PR/issue that we can link to?
cc @nrb

Reply:
I'll create one in a moment.

Reply:
Thanks for https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4865 @r4f4. Assigned myself to the upstream bug.

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4866

Reply:
Posted my reply before seeing your PR. Thanks, will review.

Reply:
Added a link to the upstream issue in the comments.